Написана лабораторная работа №2

eggorrus opened the pull request:
1. Написан заголовочный файл queue.h с xml
2. Написан cpp файл работы

eggorrus committed:
Написана лабораторная работа №2

1. Написан заголовочный файл queue.h с xml
2. Написан cpp файл работы

eggorrus committed:
Update lab2.cpp

iqqnl commented on 2sem/common/queue.h:
```diff
@@ -0,0 +1,99 @@
+class Queue
+{
+private:
+	struct Node
```
Node можно вынести отдельный файл. Код рабочий, логика очереди реализована правильно.

ichepkov2401 replied:
Крайне желательно, чтобы не дублировался код здесь и stack

iqqnl reviewed (approved):
Node можно вынести отдельный файл. Код рабочий, логика очереди реализована правильно.

ichepkov2401 commented on 2sem/common/queue.h:
```diff
@@ -0,0 +1,99 @@
+#include <stdexcept>
+
+/// <summary>
+/// Класс Queue, система хранения данных FIFO.
```
Система хранения? Очень неоднозначно

eggorrus replied:
исправил

ichepkov2401 commented on 2sem/common/queue.h:
```diff
@@ -0,0 +1,99 @@
+	};
+
+	Node* top;
+	Node* bot;
```
Полностью слова пишите, без сокращений (см. стайлгайд)

eggorrus replied:
исправлено

ichepkov2401 commented on 2sem/common/queue.h:
```diff
@@ -0,0 +1,99 @@
+		{
+			top = newData;
+			bot = newData;
+		}
```
Избыточность
![image](https://github.com/user-attachments/assets/ead963bc-65c3-4c2e-b180-61e8c38f75e4)

eggorrus replied:
устранено

ichepkov2401 commented on 2sem/lab2/lab2.cpp:
```diff
@@ -1,20 +1,118 @@
 
-// Запуск программы: CTRL+F5 или меню "Отладка" > "Запуск без отладки"
-// Отладка программы: F5 или меню "Отладка" > "Запустить отладку"
+int main()
```
Декомпозировать
![image](https://github.com/user-attachments/assets/cae802ce-6f36-4ad7-81c6-49e18dff43ee)

eggorrus replied:
Сделано

ichepkov2401 commented on 2sem/lab2/lab2.cpp:
```diff
@@ -1,20 +1,118 @@
+	Queue<int> queue;
+	int inputValue;
+	cin >> inputValue;
+	while (inputValue != 0)
```
Это по заданию не нужно. Убрать

eggorrus replied:
Убрано

ichepkov2401 commented on 2sem/lab2/lab2.cpp:
```diff
@@ -1,20 +1,118 @@
-//   4. В окне "Список ошибок" можно просматривать ошибки.
-//   5. Последовательно выберите пункты меню "Проект" > "Добавить новый элемент", чтобы создать файлы кода, или "Проект" > "Добавить существующий элемент", чтобы добавить в проект существующие файлы кода.
-//   6. Чтобы снова открыть этот проект позже, выберите пункты меню "Файл" > "Открыть" > "Проект" и выберите SLN-файл.
+		default:
```
Не хватает команд соответствующим методам структуры данных

eggorrus replied:
Исправлено

eggorrus committed:
laba2 update

1. Изменено описание класса Queue;
2. Изменены наименования полей класса Queue(избавился от сокращения bot на bottom);
3. В методе queue() класса Queue избавился от избыточности;
4.  Выполнена декомпозиция функции main
5. Добавлены новые команды, соответствующие методам структуры данных

eggorrus committed:
Убрано заполнение очереди

ichepkov2401 commented on 2sem/common/queue.h:
```diff
@@ -0,0 +1,98 @@
+class Queue
+{
+private:
+	struct Node
```
Не было внесено предложенное изменение по вынесению Node в отдельный файл или объяснено почему в данному случае этого делать не следует. Тред закрыт просто без пояснений

eggorrus replied:
Предложение по выносу в отдельный файл принято

ichepkov2401 commented on 2sem/lab2/lab2.cpp:
```diff
@@ -1,20 +1,185 @@
+	queue.queue(inputValue);
+}
+
+int removeElement()
```
Всю реализацию можно заменить на 
```
return queue.unqueue();
```
И потому непонятно зачем эта функция вообще нужна

eggorrus replied:
Функция убрана

ichepkov2401 commented on 2sem/lab2/lab2.cpp:
```diff
@@ -1,20 +1,185 @@
+	int size = queue.count();
+	int counter = 0;
+	int target;
+	cout << "Введите элемент" << endl;
```
Логику (а этот ваш метод скорее логике относится) лучше отделять от системы ввода/вывода. Тогда их можно будет изолировано переиспользовать

eggorrus replied:
Выполнено разделение логики и системы ввода/вывода

ichepkov2401 commented on 2sem/lab2/lab2.cpp:
```diff
@@ -1,20 +1,185 @@
+int main()
+{
+	setlocale(LC_ALL, "");
+	//inputQueue();
```
Это не значимый комментарий. В проде он не нужен

eggorrus replied:
убрано

ichepkov2401 commented on 2sem/lab2/lab2.cpp:
```diff
@@ -1,20 +1,185 @@
+	return element;
+}
+
+void inputQueue()
```
Метод нигде не используется

eggorrus replied:
Метод убран

eggorrus committed:
lab2_update

1. Node вынесено в отдельный файл;
2. Исправлены функции в коде работы: убрана неиспользуемая функция inputQueue и бесполезная функция removeElement;
3. В функции countNumberOfEntries выполнено разделение логики от ситсемы ввода/вывода

eggorrus committed:
Update Node.h

ichepkov2401 commented on 2sem/common/stack.h:
```diff
@@ -2,8 +2,6 @@
 #include <stdexcept>
 
-using namespace std;
-
```
Мы выносили Node чтобы не было дублирования. Следовательно из stack оно должно было уйти

eggorrus replied:
Сделано

eggorrus committed:
Update stack.h

Node.h добавлен в stack

eggorrus committed:
Устранены утечки памяти

1. в методе clear добавлено обнуление bottom;
2. добавлен блок try-catch

eggorrus committed:
Merge branch 'lab2' of https://github.com/eggorrus/2sem into lab2
